app/controllers/comments_controller.rb

```diff
@@ -1,10 +1,12 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
   before_action :is_an_authorized_user, only: [:destroy, :create, :edit]
+  before_action { authorize @comment || Comment }
```

Comment:
excellent; if you are calling the before action. Small nitpick: You do not need to add policy scope to index.